api/v1alpha1/edgeconfig_types.go

```diff
@@ -0,0 +1,185 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// EdgeConfigSpec defines the desired state of EdgeConfig
+type EdgeConfigSpec struct {
+	// Definition of the custom playbook workload to execute
+	EdgePlaybook *EdgePlaybookSpec `json:"edgePlaybook,omitempty"`
+}
+
+// EdgeConfigStatus defines the observed state of EdgeConfig
+type EdgeConfigStatus struct {
+	EdgePlaybookStatus *EdgePlaybookStatus `json:"edgePlaybookStatus,omitempty"`
+}
+
+// EdgePlaybookSpec defines the desired state of EdgePlaybook
+type EdgePlaybookSpec struct {
+	// The ansible playbook command to execute
+	AnsiblePlaybookCmd *AnsiblePlaybookCmd `json:"ansiblePlaybookCmd,omitempty"`
+}
+
+// EdgePlaybookStatus defines the observed state of EdgePlaybook
+type EdgePlaybookStatus struct {
+	Conditions []EdgePlaybookCondition `json:"conditions,omitempty"`
+}
+
+type AnsiblePlaybookCmd struct {
+	// username who execute the playbook
+	User string `json:"user,omitempty"`
+	// The ansible's playbooks list. The first element is the playbook with the highest priority.
+	// +kubebuilder:validation:MinItems=1
+	Playbooks []Playbook `json:"playbooks,omitempty"`
+}
+
+type Playbook struct {
+	// Playbook content
+	Content []byte `json:"content"`
+	// The connection timeout on ansible-playbook
+	// +kubernetes:validation:Minimum=0
+	TimeoutSeconds int64 `json:"timeoutSeconds,omitempty"`
+	// The required privelege level necessary to execute the playbook
+	RequiredPrivilegeLevel *RequiredPrivilegeLevel `json:"requiredPrivilegeLevel,omitempty"`
+	// the ansible's playbook options for the playbook
+	Options *AnsibleOptions `json:"ansibleOptions,omitempty"`
+	// the ansible's playbook privilege escalation options for the playbook
+	PrivilegeEscalationOptions *PrivilegeEscalationOptions `json:"privilegeEscalationOptions,omitempty"`
+	//Execution strategy for the playbook
+	ExecutionStrategy ExecutionStrategy `json:"playbooksPriorityMap,omitempty"`
+}
+
+type ExecutionStrategy string
+
+const (
+	StopAtFailuire ExecutionStrategy = "StopOnFailure"
+	RetryOnFailure ExecutionStrategy = "RetryOnFailure"
+	Once           ExecutionStrategy = "ExecuteOnce"
+)
+
+type AnsibleOptions struct {
+	// don't make any changes; instead, try to predict some of the changes that may occur
+	Check bool `json:"check,omitempty"`
```

Comment:
I agree that `DryRun` is better, but the ansible command is `Check` (https://docs.ansible.com/ansible/2.3/playbooks_checkmode.html). I thought to keep uniformity with ansible because it might prevent user confusion about what the option means. What are your thoughts on this?

Reply:
To make the decision harder, they mention both options (check and dry-run) in that link :-) 
If the convention of the CR to follow ansible options naming, then let's keep it as is.